src/lib/array.jl

```diff
@@ -106,17 +101,6 @@ end
 
 @adjoint collect(x::Array) = collect(x), Δ -> (Δ,)
 
-@adjoint fill(x::Real, dims...) = fill(x, dims...), Δ->(sum(Δ), map(_->nothing, dims)...)
-
-@adjoint function circshift(A, shifts)
-  circshift(A, shifts), Δ -> (circshift(Δ, map(-, shifts)), nothing)
-end
-
-@adjoint function reverse(x::AbstractArray, args...; kwargs...)
```

Comment:
Ditto here - this is needed for bidirectional recurrent cells

Reply:
Added in 1.5: https://github.com/JuliaDiff/ChainRules.jl/blob/7593339839832ffb7ac83401918390ffe9d4eb42/src/rulesets/Base/array.jl#L314

The tests appear to pass -- are there not tests for `reverse` in Zygote?

Reply:
Ah we are still testing, excellent?